src/shim.ts

```diff
@@ -6,43 +23,220 @@ import {
     const listener = new Orchestrator(fn).listen();
-    return async (context, orchestrationTrigger): Promise<OrchestratorState> => {
+
+    return async (
+        orchestrationTrigger: DurableOrchestrationInput,
+        context: IOrchestrationFunctionContext
+    ): Promise<OrchestratorState> => {
         return await listener(context, orchestrationTrigger);
     };
 }
 
-export function entity<T = unknown>(
-    fn: (context: IEntityFunctionContext<T>) => void
-): (
-    context: IEntityFunctionContext<T>,
-    entityTrigger: DurableEntityBindingInfo
-) => Promise<EntityState> {
+/**
+ * Enables an entity handler function to act as a Durable Entity Azure Function.
+ *
+ * @param fn the handler function that should act as a durable entity
+ */
+export function createEntityFunction<T = unknown>(fn: EntityHandler<T>): EntityFunction<T> {
     const listener = new Entity<T>(fn).listen();
-    return async (context, entityTrigger): Promise<EntityState> => {
+
+    return async (
+        entityTrigger: DurableEntityBindingInfo,
+        context: IEntityFunctionContext<T>
+    ): Promise<EntityState> => {
         return await listener(context, entityTrigger);
     };
 }
+
+export namespace app {
+    /**
+     * Registers a generator function as a Durable Orchestrator for your Function App.
+     *
+     * @param functionName the name of your new durable orchestrator
+     * @param handler the generator function that should act as an orchestrator
+     *
+     * @example Register an orchestrator
+     * ```javascript
+     * const df = require("durable-functions");
+     *
+     * df.orchestration('durableOrchestration1', function* (context) {
+     *     // orchestrator body
+     * });
+     * ```
+     */
+    export function orchestration(functionName: string, handler: OrchestrationHandler): void;
+
+    /**
+     * Registers a generator function as a Durable Orchestrator for your Function App.
+     *
+     * @param functionName the name of your new durable orchestrator
+     * @param options the configuration options object describing the handler for this orchestrator
+     *
+     * @example Register an orchestrator
+     * ```javascript
+     * const df = require("durable-functions");
+     *
+     * df.orchestration('durableOrchestration1', {
+     *   handler: function* (context) {
+     *       // orchestrator body
+     *   }
+     * });
+     * ```
+     */
+    export function orchestration(functionName: string, options: OrchestrationOptions): void;
+
+    export function orchestration(
+        functionName: string,
+        handlerOrOptions: OrchestrationHandler | OrchestrationOptions
+    ): void {
+        const handler: OrchestrationHandler =
+            typeof handlerOrOptions === "function" ? handlerOrOptions : handlerOrOptions.handler;
+
+        azFuncApp.generic(functionName, {
+            trigger: trigger.orchestration(),
+            handler: createOrchestrator(handler),
+        });

```

Comment:
We want to pass along the user's options, so it should be something like this:
```typescript
const options: OrchestrationOptions =
    typeof handlerOrOptions === "function"
        ? { handler: handlerOrOptions }
        : handlerOrOptions;

azFuncApp.generic(functionName, {
    trigger: trigger.orchestration(),
    ...options,
    handler: createOrchestrator(options.handler),
});
```

Reply:
Done